### PR TITLE
ssl: do not choose auto DH groups that are weaker than the security level

### LINKED ISSUE #15808: DHE-PSK-* ciphersuites fail when SECLEVEL requires key sizes above 1024
When I try to establish a connection with TLS 1.2 and one of the DHE-PSK-* ciphers, the _server_ aborts it with the message:
```
4087A703F27F0000:error:0A00018A:SSL routines:tls_construct_server_key_exchange:dh key too small:ssl/statem/statem_srvr.c:2484:
```

That's because it loads 1024 bit DH parameters by default.

At the very least, I think the default parameters should be increased to 2048 bit.
I'm not sure if the server should query the minimal key size to select DH parameters so that it continues to work in the future...

Using master f77208693ec3bda99

Reproducer:
```
openssl s_server  -cipher @SECLEVEL=2:DHE-PSK-AES128-CCM -nocert -psk 0102030405 -www
```
and on separate console:
```
openssl s_client -connect localhost:4433 -cipher DHE-PSK-AES128-CCM  -psk 0102030405 -no_tls1_3
```

### PR DESCRIPTION
**Pull request by paulidale**
I'm not sure if this is sensible for 1.1.1 or not.

Fixes #15808

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: test/recipes/80-test_ssl_old.t [hunk 1]
```diff
@@ -549,14 +549,17 @@ sub testssl {
...
+	    skip "skipping PSK tests", 3
 	        if ($no_psk);
 
 	    ok(run(test([@ssltest, "-tls1", "-cipher", "PSK", "-psk", "abc123"])),
 	       'test tls1 with PSK');
 
 	    ok(run(test([@ssltest, "-bio_pair", "-tls1", "-cipher", "PSK", "-psk", "abc123"])),
 	       'test tls1 with PSK via BIO pair');
+
+            ok(run(test(['ssl_old_test', '-psk', '0102030405', '-cipher', '@SECLEVEL=2:DHE-PSK-AES128-CCM'])),
```
Comment (mattcaswell) [inline test/recipes/80-test_ssl_old.t:561]: Indentation seems off here

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: test/recipes/80-test_ssl_old.t (hunk 1 above)
Comment (paulidale) [reply test/recipes/80-test_ssl_old.t:561]: Spaces vs tabs.  I've used spaces, the file has tabs.  The indentation looks fine when editing.

Comment (paulidale) [reply test/recipes/80-test_ssl_old.t:561]: I've pushed a de-tabbing commit.

Comment (paulidale) [reply test/recipes/80-test_ssl_old.t:561]: I reverted the de-tabbing in favour of #15824

**Review by mattcaswell — APPROVED** [review]
Approved on the basis that #15824 will sort out the tabbing issues.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
As for the 1.1.1 backport - IMO this can be seen as a bug fix because the server with that automatically chosen group really does not work. It also effectively affects only PSK-only servers because with servers having a certificate a higher security level means that the certificate has to be strong enough to support that level and that implies stronger DH parameters even without this fix. So I would be OK with the backport.

**Comment by t8m** [comment]
(I suppose this won't cherry-pick cleanly though so it will need a separate PR anyway.)

**Comment by paulidale** [comment]
Merged to master, thanks for the feedback.

**Pull request closed without merging**